examples/vision/zero_dce.py

```diff
@@ -0,0 +1,444 @@
+    conv4 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(conv3)
+    int_con1 = layers.Concatenate(axis=-1)([conv4, conv3])
+    conv5 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(int_con1)
+    int_con2 = layers.Concatenate(axis=-1)([conv5, conv2])
+    conv6 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(int_con2)
+    int_con3 = layers.Concatenate(axis=-1)([conv6, conv1])
+    x_r = layers.Conv2D(24, (3, 3), strides=(1, 1), activation="tanh", padding="same")(
+        int_con3
+    )
+    return keras.Model(inputs=input_img, outputs=x_r)
+
+
+"""
+## Loss Functions
+
+To enable zero-reference learning in DCE-Net, we would use a set of differentiable
+non-reference losses that allow us to evaluate the quality of enhanced images.
+"""
+
+"""
+### Color Constancy Loss
+
+The Color Constancy Loss is used to correct the potential color deviations in the
+enhanced image and also build the relations among the three adjusted channels.
+"""
+
+
+def color_constancy_loss(x):
+    mean_rgb = tf.reduce_mean(x, [1, 2], keepdims=True)
+    mr, mg, mb = mean_rgb[:, :, :, 0], mean_rgb[:, :, :, 1], mean_rgb[:, :, :, 2]
+    d_rg = tf.pow(mr - mg, 2)
+    d_rb = tf.pow(mr - mb, 2)
+    d_gb = tf.pow(mb - mg, 2)
+    return tf.pow(tf.pow(d_rg, 2) + tf.pow(d_rb, 2) + tf.pow(d_gb, 2), 0.5)
+
+
+"""
+### Exposure loss
+
+To restrain under-/over-exposed regions, the Exposure Control Loss is used. The exposure
+control loss measures the distance between the average intensity value of a local region
+to the well-exposedness level which is set to `0.6`.
+"""
+
+
+def exposure_loss(x, mean_val=0.6):
+    x = tf.reduce_mean(x, 3, keepdims=True)
+    mean = tf.keras.layers.AveragePooling2D(pool_size=16, strides=16)(x)
+    return tf.reduce_mean(tf.pow(mean - mean_val, 2))
+
+
+"""
+### Illumination Smoothness Loss
+
+To preserve the monotonicity relations between neighboring pixels, the Illumination
+Smoothness Loss is added to each curve parameter map.
+"""
+
+
+def illumination_smoothness_loss(x):
+    batch_size = tf.shape(x)[0]
+    h_x = tf.shape(x)[1]
+    w_x = tf.shape(x)[2]
+    count_h = (tf.shape(x)[2] - 1) * tf.shape(x)[3]
+    count_w = tf.shape(x)[2] * (tf.shape(x)[3] - 1)
+    h_tv = tf.reduce_sum(tf.pow((x[:, 1:, :, :] - x[:, : h_x - 1, :, :]), 2))
+    w_tv = tf.reduce_sum(tf.pow((x[:, :, 1:, :] - x[:, :, : w_x - 1, :]), 2))
+    batch_size = tf.cast(batch_size, dtype=tf.float32)
+    count_h = tf.cast(count_h, dtype=tf.float32)
+    count_w = tf.cast(count_w, dtype=tf.float32)
+    return 2 * (h_tv / count_h + w_tv / count_w) / batch_size
+
+
+"""
+### Spatial Consistancy Loss
+
+The spatial consistency loss encourages spatial coherence of the enhanced image through
+preserving the difference of neighboring regions between the input image and its
+enhanced version.
+"""
+
+
+class SpatialConsistancyLoss:
```

Comment:
This loss matches the expected signature `y_true, y_pred` so prefer subclass the `Loss` class

Reply:
Changed `SpatialConsistancyLoss` to a subclass of `keras.losses.Loss` and changed the `__call__(original, enhanced)` signature to `call(y_true, y_pred)`.